base/save.go

```diff
@@ -240,3 +246,39 @@ func GenerateAvailableName(r repo.Repo, peername, prefix string) string {
 		}
 	}
 }
+
+// Drop sets named components to nil
+func Drop(ds *dataset.Dataset, revStr string) error {
+	if revStr == "" {
+		return nil
+	}
+
+	revs, err := dsref.ParseRevs(revStr)
+	if err != nil {
+		return err
+	}
+
+	for _, rev := range revs {
+		if rev.Gen != 1 {
+			return fmt.Errorf("cannot drop specific generations")
+		}
+		switch rev.Field {
```

Comment:
Could we have this live in `base/component` like https://github.com/qri-io/qri/blob/ef62c71198c2110b5d4de5c5cf7245c78d8a9aed/base/component/field_id.go
And tag these for this specific usecase.

Reply:
don't think it should move into the component package. We haven't fully documented this, but the component package handles dynamic behavior on components using a new internal API. If anything those values `field_id.go` should move into the `dsref` package.

That said, I'm very much with you, `Drop` shouldn't be in `save.go`, so I've moved it to `base/revisions.go` and added tests.